app/models/mixins/service_mixin.rb

```diff
@@ -116,12 +96,37 @@ def next_group_index(current_idx, direction = 1)
 
     (current_idx + direction).send(method, target) do |i|
       next if i == current_idx
-      return(i) if self.group_has_resources?(i)
+      return(i) if group_has_resources?(i)
     end
 
     nil
   end
 
+  private
+
+  def enforce_single_service_parent(resource)
+    if resource.try(:enforce_single_service_parent?) == true && resource.service
+      raise MiqException::Error, _("<%{class_name}> <%{id}>:<%{name}> is already connected to a service.") %
+                                 {:class_name => resource.class.name, :id => resource.id, :name => resource.name}
+    end
+  end
+
+  def create_service_resource(rsc, options)
+    if circular_reference?(rsc)
```

Comment:
@gmcculloug should we have a single validate function that can check for single parent service and circular reference and 
``` ruby
   def validate(resource)
        enforce_single_service_parent(resource)
        if circular_reference?(resource)
          raise MiqException::MiqServiceCircularReferenceError,
             _("Adding resource <%{resource_name}> to Service <%{name}> will create a circular reference") %
             {:resource_name => resource.name, :name => name}
         end
    end
```

Reply:
@mkanoor I agree this could be a nice change but I would prefer to look into refactoring this is a separate PR.